src/lib.rs

```diff
@@ -517,6 +530,25 @@ fn pan_orbit_camera(
         }
 
         // 4 - Update the camera's transform based on current values
+        pan_orbit.radius = pan_orbit.radius.map(|radius| {
+            if pan_orbit.target_radius != radius {
```

Comment:
I'd like to move some code around so the radius logic follows the same pattern as alpha/beta. Which involves the following:

1. After line 498, set `has_moved = true`
2. Modify line 553 to include radius, i.e. ` if let (Some(alpha), Some(beta), Some(radius)) = (pan_orbit.alpha, pan_orbit.beta, pan_orbit.radius) {`
3. Add `pan_orbit.target_radius != radius` to the existing `if` condition on line 554, and move this block of code into that if block.

Doing this clearly separates getting input from the user, and applying that to the camera. And following the same pattern makes everything easier to understand as a whole.